visualization/app/codeCharta/ui/codeMap/codeMap.mouseEvent.service.ts

```diff
@@ -259,24 +262,11 @@ export class CodeMapMouseEventService implements ViewCubeEventPropagationSubscri
 	}
 
 	private drawTemporaryLabelFor(codeMapBuilding: CodeMapBuilding, labels: Object3D[]) {
-		const appSettings = this.storeService.getState().appSettings
-		const showLabelNodeName = appSettings.showMetricLabelNodeName
-		const showLabelNodeMetric = appSettings.showMetricLabelNameValue
+		const { showMetricLabelNodeName, showMetricLabelNameValue } = this.storeService.getState().appSettings
 
-		let displayLabelMetricName = true
+		const displayLabelMetricName = !showMetricLabelNameValue || showMetricLabelNodeName
```

Comment:
Enforcement is only necessary when showMetricLabelNameValue and showMetricLabelNodeName are disabled.

```suggestion
		const enforceLabelMetricName = !showMetricLabelNameValue && !showMetricLabelNodeName ? true : false
```

Reply:
I guess your version might be easier to follow due to focusing on the enforced part instead of when to display the metric name. Returning `true` or `false` explicitly is not required though. That's already the case due to the statement itself being a boolean. :-)

Reply:
I see. I leave it up to you whether you want to change it or not.

Reply:
I refactored the code a bit more so that the label enforcing becomes clearer.